proof/crefine/RISCV64/Refine_C.thy

```diff
@@ -892,17 +892,20 @@ lemma dmo_domain_user_mem'[wp]:
   done
 
 lemma do_user_op_corres_C:
-  "corres_underlying rf_sr False False (=) (invs' and ex_abs einvs) \<top>
-                     (doUserOp f tc) (doUserOp_C f tc)"
+  "corres_underlying rf_sr False False (=)
+     (invs' and (\<lambda>s. sch_act_wf (ksSchedulerAction s) s) and ex_abs einvs) \<top>
+     (doUserOp f tc) (doUserOp_C f tc)"
   apply (simp only: doUserOp_C_def doUserOp_def split_def)
   apply (rule corres_guard_imp)
     apply (rule_tac P=\<top> and P'=\<top> and r'="(=)" in corres_split)
        apply (clarsimp simp: simpler_gets_def getCurThread_def
                 corres_underlying_def rf_sr_def cstate_relation_def Let_def)
-      apply (rule_tac P=invs' and P'=\<top> and r'="(=)" in corres_split)
+      apply (rule_tac P="invs' and (\<lambda>s. sch_act_wf (ksSchedulerAction s) s)"
+                  and P'=\<top> and r'="(=)" in corres_split)
```

Comment:
I can't remember, did we ever make a decision about how we want to format and indent multi-line rule_tacs? I have a pretty strong preference for the `in ...` part to at least be on its own line, as in my mind it is more important than the variables being instantiated.

Reply:
I'd say this one is admissible, but I'd not object to putting the `in ..` part on a separate line.